### DISCUSSION TITLE
How to just get the data, don't need fetch again

### DISCUSSION THREAD

**Discussion by IRediTOTO** [vercel/swr#753] [Q&A]
Hi, I was fetch data in other component 

```
 await fetch("/api/search?data=" + e.target.value, {
               method: "POST",
            })
               .then((res) => res.json())
               .then((res) => {
                  mutate("searchResult", res, false);
                  return res;
               })
```

Now I am trying get data but don't need to fetch again
  ` const { data } = useSWR("searchResult", null,{revalidateOnFocus:false,revalidateOnMount:false,revalidateOnReconnect:false,shouldRetryOnError:false});`

But swr still fetch("searchResult")

Is there a good way to stop it ?

**Comment by promer94** [comment]
when you use swr like ```const { data } = useSWR("searchResult", null)```. It would never fire any request and only rerender when you use mutate to change the cache data.  Here is a simple [example](https://codesandbox.io/s/mutate-example-1vevk?file=/src/App.js)

However, this kind of usage is not well documented and probably will be deprecated in the future.  FYI #738

**Reply by IRediTOTO** [reply]
I tried, it will but ok, I added some option and it will not fetch. I am trying use it like a fast React Context. It's good I think, why remove it? @@

**Reply by IRediTOTO** [reply]
```
const { data } = useSWR("searchResult", null, {
      revalidateOnFocus: false,
      revalidateOnMount: false,
      revalidateOnReconnect: false,
      shouldRetryOnError: false,
   });
```
I must use like this :D

**Reply by IRediTOTO** [reply]
Is there any func like this? 
import {getData} from "swr"
...
const whatINeed= getData("cookKeyData")

**Reply by shuding** [reply]
> Is there any func like this?
> import {getData} from "swr"

Nope there isn't. The issue is the data might get staled when you want to process it. So the best way would be getting the `data` inside a component, which reads from the `useSWR` hook (always fresh data).
